rclcpp/test/test_time.cpp

```diff
@@ -48,3 +50,90 @@ TEST(TestTime, rate_basics) {
   EXPECT_NE(0, default_now.sec);
   EXPECT_NE(0u, default_now.nanosec);
 }
+
+TEST(TestTime, convertions) {
+  rclcpp::Time now = rclcpp::Time::now();
+  builtin_interfaces::msg::Time now_msg = now;
+
+  rclcpp::Time now_again = now_msg;
+  EXPECT_EQ(now.nanoseconds(), now_again.nanoseconds());
+
+  builtin_interfaces::msg::Time msg;
+  msg.sec = 12345;
+  msg.nanosec = 67890;
+
+  rclcpp::Time time = msg;
+  EXPECT_EQ(
+    RCL_S_TO_NS(static_cast<uint64_t>(msg.sec)) + static_cast<uint64_t>(msg.nanosec),
+    time.nanoseconds());
+  EXPECT_EQ(static_cast<uint64_t>(msg.sec), RCL_NS_TO_S(time.nanoseconds()));
+
+  builtin_interfaces::msg::Time negative_time_msg;
+  negative_time_msg.sec = -1;
+  negative_time_msg.nanosec = 1;
+  try {
+    rclcpp::Time negative_time = negative_time_msg;
+    FAIL();
+  } catch (const std::exception &) {
+    SUCCEED();
+  }
```

Comment:
If the invoked code is supposed to raise a specific exception I would suggest to use `EXPECT_THROW` rather than `EXPECT_ANY_THROW`.